engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/bool/CompareToNode.java

```diff
@@ -29,11 +29,11 @@ Atom doBoolean(Boolean self, Boolean that) {
     }
   }
 
-  @Specialization
-  Atom doOther(Boolean self, Object that) {
+  @Fallback
+  Object doOther(Boolean self, Object that) {
     CompilerDirectives.transferToInterpreter();
     var bool = Context.get(this).getBuiltins().bool().getType();
-    var typeError = Context.get(this).getBuiltins().error().makeTypeError(that, bool, "that");
-    throw new PanicException(typeError, this);
+    var typeError = Context.get(this).getBuiltins().error().makeTypeError(bool, that, "that");
+    return DataflowError.withoutTrace(typeError, this);
```

Comment:
Raising this error without any trace will make it harder to debug.
While I appreciate that in many cases we want to avoid adding traces to Dataflow errors for performance, if we are shifting towards using dataflow errors much more than panics, I think we need to revisit that decision.

Maybe we should have 2 classes of errors? Cheap indicator errors that can happen in hot computations that need to be without trace for performance, and the more expensive 'real errors' that used to most often used to be panics but were shifted to dataflow. For the latter 'class', I'd add traces so we can find out where the error has happened.

Reply:
Producing stacktrace is expensive. However including `getSourceSection()` - e.g. location in the source should be cheap - if that's enough. The `DataflowError` has reference to `this` - it can use `this.getSourceSection()` later to construct where the error comes from and that shall be fast.

Reply:
Interesting point.
Still for the more crucial errors I'd consider a full stacktrace as it can give much more information.
Since getting just the source location is so cheap, maybe we should include this for _all_ dataflow errors? That could be quite helpful for debugging still.

Reply:
When I tested the code the `getSourceSection` didn't appear to work - I think it would be a good addition to the Dataflow error so suggest we add a task to pick this up later.